infrastructure/src/main/java/postech/soat/tech/challenge/api/controller/ProductController.java

```diff
@@ -0,0 +1,39 @@
+package postech.soat.tech.challenge.api.controller;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.*;
+import postech.soat.tech.challenge.api.request.ProductDTO;
+import postech.soat.tech.challenge.api.request.ProductRequestMapper;
+import postech.soat.tech.challenge.api.response.ApiResponse;
+import postech.soat.tech.challenge.model.Product;
+import postech.soat.tech.challenge.port.input.CreateProductUseCase;
+import postech.soat.tech.challenge.port.input.FindProductUseCase;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/api/produto")
+public class ProductController {
+
+    private final CreateProductUseCase createProductUseCase;
+    private final FindProductUseCase findProductUseCase;
+
+    public ProductController(CreateProductUseCase createProductUseCase, FindProductUseCase findProductUseCase) {
+        this.createProductUseCase = createProductUseCase;
+        this.findProductUseCase = findProductUseCase;
+    }
+
+    @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
+    public ApiResponse<Product> createProduto(@RequestBody ProductDTO productDTO) {
+        Product createdProduct = createProductUseCase.createProduct(ProductRequestMapper.INSTANCE.toNewProduct(productDTO));
```

Comment:
Por que usar uma interface para o mapper? Digo, por que não criar métodos de mapping no próprio DTO?

Pensei em algo como:

```suggestion
        Product createdProduct = createProductUseCase.createProduct(productDTO.toProduct());
```

E o caminho contrário poderia ser algo como:
```java
ProductDTO.fromProduct(product);
```
_(se tivessemos extensiom methods em java, seria perfeito aqui!)_

Reply:
Gosto da ideia sim. A principio pensei em suar o MapStruct, mas ele agrega tão pouco que acho que não compensa. Vou mudar.